SwiftDungeon/Player.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Player.swift
+//  SwiftDungeon
+//
+//  Created by Puntillo Andrew J. on 3/12/18.
+//  Copyright © 2018 Toro Juan D. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+class Player : Entity {
+    
+    //The direction the player will move towards to on input
+    var direction:CGPoint = CGPoint(x:0, y:0)
+    //Speed of player
+    private let velocity: CGFloat = 10
+    //Animation
+    private var textureAnimation: [SKTexture] = []
+    
+    init() {
+        super.init(imageName: "rogue_idle_01")
+        idle()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func update(_ currentTime: TimeInterval) {
+        super.update(currentTime)
+        if (direction.x != 0 || direction.y != 0) {
+            moveTo()
+        }
+        position.x -= velocity * direction.x
+        position.y += velocity * direction.y
+
+    }
+    
+    //Idle animation
+    func idle() {
+        textureAnimation = [SKTexture(imageNamed: "rogue_idle_01"),
```

Comment:
If you move the textures to a static variable inside your class they will only need to be created once and you can pass them as needed to your actions.